src/renderer/components/item-object-list/item-list-layout-content/item-list-layout-content.tsx

```diff
@@ -0,0 +1,271 @@
+  copyClassNameFromHeadCells?: boolean;
+  sortingCallbacks?: TableSortCallbacks<I>;
+  tableProps?: Partial<TableProps<I>>; // low-level table configuration
+  renderTableHeader: TableCellProps[] | null;
+  renderTableContents: (item: I) => (ReactNode | TableCellProps)[];
+  renderItemMenu?: (item: I, store: ItemStore<I>) => ReactNode;
+  customizeTableRowProps?: (item: I) => Partial<TableRowProps>;
+  addRemoveButtons?: Partial<AddRemoveButtonsProps>;
+  virtual?: boolean;
+
+  // item details view
+  hasDetailsView?: boolean;
+  detailsItem?: I;
+  onDetails?: (item: I) => void;
+
+  // other
+  customizeRemoveDialog?: (selectedItems: I[]) => Partial<ConfirmDialogParams>;
+
+  /**
+   * Message to display when a store failed to load
+   *
+   * @default "Failed to load items"
+   */
+  failedToLoadMessage?: React.ReactNode;
+}
+
+@observer
+export class ItemListLayoutContent<I extends ItemObject> extends React.Component<ItemListLayoutContentProps<I>> {
+  constructor(props: ItemListLayoutContentProps<I>) {
+    super(props);
+    makeObservable(this);
+  }
+
+  @computed get failedToLoad() {
+    return this.props.store.failedLoading;
+  }
+
+  @boundMethod
+  getRow(uid: string) {
+    const {
+      isSelectable, renderTableHeader, renderTableContents, renderItemMenu,
+      store, hasDetailsView, onDetails,
+      copyClassNameFromHeadCells, customizeTableRowProps, detailsItem,
+    } = this.props;
+    const { isSelected } = store;
+    const item = this.props.getItems().find(item => item.getId() == uid);
+
+    if (!item) return null;
+    const itemId = item.getId();
+
+    return (
+      <TableRow
+        key={itemId}
+        nowrap
+        searchItem={item}
+        sortItem={item}
+        selected={detailsItem && detailsItem.getId() === itemId}
+        onClick={hasDetailsView ? prevDefault(() => onDetails(item)) : undefined}
+        {...customizeTableRowProps(item)}
+      >
+        {isSelectable && (
+          <TableCell
+            checkbox
+            isChecked={isSelected(item)}
+            onClick={prevDefault(() => store.toggleSelection(item))}
+          />
+        )}
+        {
+          renderTableContents(item).map((content, index) => {
+            const cellProps: TableCellProps = isReactNode(content) ? { children: content } : content;
+            const headCell = renderTableHeader?.[index];
+
+            if (copyClassNameFromHeadCells && headCell) {
+              cellProps.className = cssNames(cellProps.className, headCell.className);
+            }
+
+            if (!headCell || this.showColumn(headCell)) {
+              return <TableCell key={index} {...cellProps} />;
+            }
+
+            return null;
+          })
+        }
+        {renderItemMenu && (
+          <TableCell className="menu">
+            <div onClick={stopPropagation}>
+              {renderItemMenu(item, store)}
+            </div>
+          </TableCell>
+        )}
+      </TableRow>
+    );
+  }
+
+  @boundMethod
```

Comment:
I'd like to remove `@boundMethod` in nearest future since we currently have 2 libraries for that:


```ts
// lens/src/common/utils/autobind.ts
import { boundMethod, boundClass } from "autobind-decorator";
import autoBindClass, { Options } from "auto-bind";...
```

For most cases to bind all class methods we use this:
```ts
constructor(private dependencies: Dependencies)
    makeObservable(this);
    autoBind(this); // note: this is important to call after `makeObservable`
```

For binding single methods I'm suggesting stick to arrow-functions or to use re-assign class fields with `.bind(this)` inside constructor.

Reply:
I'd like to remove all these binds. But, this is old code and not touching this on this PR